packages/devtools_app/test/flutter/table_test.dart

```diff
@@ -161,6 +161,20 @@ void main() {
       expect(find.byKey(const Key('empty')), findsOneWidget);
     });
 
+    testWidgets('displays with multiple data roots',
```

Comment:
Please also add a test for interactions on the roots: ensure that a few of the roots are properly expandable without breaking the state of the rest of the tree.

Reply:
Also, include tests for the cases where the roots of the tree are changed.  We don't want to get a red screen of death when we request new data and we get a broken tree because there were fewer roots being used by the same tree representation.

Reply:
done. ptal

Reply:
Very nice!
![dicaprio](https://cloud.githubusercontent.com/assets/14947366/18679346/18517596-7f14-11e6-81e1-548dc7901061.gif)